janitor/functions/utils.py

```diff
@@ -243,10 +243,13 @@ def _column_sel_dispatch(columns_to_select, df):  # noqa: F811
     if _is_str_or_cat(df_columns):
         if columns_to_select in df_columns:
             return [columns_to_select]
-        outcome = fnmatch.filter(df_columns, columns_to_select)
-        if not outcome:
+        # fix for Github Issue 1160
+        outcome = [
+            fnmatch.fnmatchcase(column, columns_to_select) for column in df
+        ]
+        if not any(outcome):
             raise KeyError(f"No match was returned for '{columns_to_select}'.")
-        return outcome
+        return df_columns[outcome].tolist()
```

Comment:
I thought we could directly return `outcome`, because of `outcome == df_columns[outcome].tolist()`